src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -1514,6 +1519,48 @@ def test_group_deployment_crosstenant(self, resource_group):
             self.check('resourceGroup', '{vm_rg}')
         ])
 
+        self.cmd('group deployment create -g {vm_rg} -n {dn1} --template-file "{tf}" --parameters SIG_ImageVersion_id={sig_id} NIC_id={nic_id} --aux-tenants "{aux_tenant}"', checks=[
```

Comment:
Can you consider abstracting this part to a new test function?

Reply:
This question has been considered before, my thoughts are: 
The name of this method is  `test_group_deployment_crosstenant()`, but the original test was only for cross aux_sub. 
Now I've added the test logic for cross aux_tenant, which is consistent with the method name.
My original plan was to directly delete the logic of testing aux_sub in `test_group_deployment_crosstenant()` after deprecating `--aux-sub`. So that some common code can be reused and the test with two parameters bound together can be implemented properly.
What do you think about it?

Reply:
And I will add a new test function for the refactored code, so I think it might be better to use only a function for the old code to distinguish the tests of the old code from the new code.